endpoint.go

```diff
@@ -804,9 +810,27 @@ func (ep *endpoint) assignAddressVersion(ipVer int, ipam ipamapi.Ipam) error {
 		return nil
 	}
 
+	// Validate preferred address
+	if prefAdd != nil {
+		for _, d := range ipInfo {
```

Comment:
At first I was not able to combine both things in the following loop in a clean way.
Let me give it one more try.